zk/l1_data/l1_decoder.go

```diff
@@ -65,9 +88,39 @@ func DecodeL1BatchData(txData []byte) ([][]byte, common.Address, error) {
 	if err != nil {
 		return nil, coinbase, err
 	}
-	err = json.Unmarshal(bytedata, &sequences)
-	if err != nil {
-		return nil, coinbase, err
+
+	if !isValidium {
+		err = json.Unmarshal(bytedata, &sequences)
+		if err != nil {
+			return nil, coinbase, err
+		}
+	} else {
+		var validiumSequences []ValidiumBatchData
```

Comment:
I would extract it to a method, already many lines of logic. Ideally even have something like that

```
if isValidium {
    GetSequencesForValidum(…)
} else {
    GetSequencesForRollup(…)
}
```

Reply:
Sounds good, will do.